### PR TITLE
Extend and tweak sidebar "structured"

### PR DESCRIPTION
**Pull request by mlange-42**
#### Summary
Interface "Extend and tweak sidebar 'structured'"

#### Purpose of change

Make sidebar `structured` more useful and maintainable.

#### Describe the solution

* [x] Split sidebar JSON into multiple files in sub-folder `structured` (d297dc5, cc53078)
* [x] Add wielded weapon above compass, as this place became free with #59442 (76a0bdd)
* [x] Add widget for MA buffs (disabled by default) (32c6915)
* [x] Add a version of the health widget that shows the overmap instead of a bodygraph (79b6d7f)
* [x] Add help text to all top-level widgets (9ffa396, 1f14aef)
* [x] Make a widget that composes weapon, ammo and mode from multiple widgets, for better display of guns with long names (64515ac)
   * Split function `Character::weapname()` into functions for parts, only merge parts in function `weapname()`
   * Make name parts (simple name, mode, ammo) available for widgets
* [x] Add a mana widget for Magiclysm (197f39c)
* [x] \o/ Add health widget variant with all four bodygraphs like in #59454 (2d18988)
* [x] ~~Merge all three environment widgets into one (requires #59676)~~ Will do in a separate PR after merge of the dependency

#### Describe alternatives you've considered

None

#### Testing

Tested all widgets in-game

#### Additional context

Note: PR author is the original author of the sidebar.

Screenshot of the sidebar, showing most of the new widgets and changes. Note that mana and the martial arts row are disabled by default.

Screenshot of sidebar settings, depicting an exemplary widget help text.

### CODE REVIEW

**Comment by mlange-42** [comment]
Label `Code: Tests` can be removed. Rebased to `master` after merge of #59571, so the changes to the tests are no longer in this PR.

**Comment by mlange-42** [comment]
@TadhgEomonn FYI: Mana/Magiclysm support added.

**Comment by Rjtheperson** [comment]
Updated to the latest experimental and loading my save file gives me this.
image
If I ignore this error it just shoots me back to the main menu, any advice?

**Comment by mlange-42** [comment]
@Rjtheperson Do you have the old mod for the sidebar installed?

Otherwise, it might be caused by the fact that the sidebar file is in a sub-folder now. Go to `data/json/ui`, and delete `sidebar-structured.json`. There should also be a folder `structured` (the new location, leave as is). 

Maybe the Launcher (?) did not delete the moved file. Would be bad, as it would affect a lot of users.

File: src/character.cpp [hunk 1]
```diff
@@ -7192,18 +7233,15 @@ std::string Character::weapname() const
...
 
-        return string_format( "%s%s%s", gunmode, gun_name, mag_ammo );
-
-    } else if( !is_armed() ) {
-        return _( "fists" );
+        return mag_ammo;
 
     } else {
-        return weapon.tname();
+        return _( "" );
```
Comment (BevapDin) [inline src/character.cpp:7244]: Is it really necessary to translate an empty string? How should it be translated anyway?

**Pull request merged**